requests/__init__.py

```diff
@@ -48,11 +48,20 @@
 __copyright__ = 'Copyright 2016 Kenneth Reitz'
 
 # Attempt to enable urllib3's SNI support, if possible
-try:
-    from .packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
-except ImportError:
-    pass
+# If the user has a recent version of OpenSSL, this
+# isn't needed as Python's core ssl module handles SNI, etc correctly.
+# If the requirements of urllib3.contrib.pyopenssl change,
+# this version should be modified appropraitely.
+#
+# See https://github.com/kennethreitz/requests/issues/3213 as well.
+import ssl
+
+if ssl.OPENSSL_VERSION_INFO < (1, 0, 1):
```

Comment:
What do you perceive as the benefit of doing a version check here? To verify what version of cryptography you've got you need to attempt to import it. If it imports successfully and the version is new enough then you would attempt to import pyOpenSSL, but we already effectively do that. I suppose a version check could potentially catch a mismatch between cryptography version and the minimum required by pyOpenSSL (since pip doesn't do much in the way of version resolution this can occur in some scenarios), but that check would more properly be part of pyOpenSSL.

The check present in the currently vendored version of urllib3 is insufficiently robust to catch the scenario where pyOpenSSL < 0.14 was present, but that's been updated (see: https://github.com/shazow/urllib3/pull/1063) and will be in a future requests.